src/ExpressionParser.php

```diff
@@ -432,6 +432,52 @@ public function parseMappingExpression()
         return $node;
     }
 
+    /*
+     * Unlike its sibling methods above, method requires a TokenStream parameter.
```

Comment:
I vote for making this method work like the others instead.

Reply:
I would even suggest removing that method entirely by moving this logic to the TypesTokenParser instead. This would be cleaner by keeping the ExpressionParser responsible for parsing expressions (while the TypesTokenParser explicitly expects a different syntax than a Twig expression)

Reply:
First off, thanks for the review! 🙇

> I vote for making this method work like the others instead.

I tried that at first, but I couldn't write tests for it like the others (e.g., `testSequenceExpression`) because the `ExpressionParser` never calls this new method directly.

> I would even suggest removing that method entirely by moving this logic to the TypesTokenParser instead. 

Yeah, that's what I had initially. @ruudk also agrees that its better (for now) to move this (back) to `TypesTokenParser`.

I'll `rebase -i` my branch accordingly.